### ISSUE TITLE
Oh I see this is kind of like you branch for rendering a web page.

### ISSUE THREAD

**Issue by russelljjarvis** [TsaiHuiAnn/TsaiHuiAnn.github.io#1]

**Comment by russelljjarvis** [comment]
On the command line on your computer do something like:
Navigate to the repositories directory:
```
git branch gh-pages
```

Here it might complain that certain file changes need to be committed or they will be overwritten. Its important that you learn to understand how to do:

```
git status
```

`git add filename.filetype`

`git commit -m "your commit message" `

From here it should let you run # 1, but probably run # 2 for your self, not because you have to.

# 2 `git push origin master`

# 1
```
git checkout gh-pages
```

# 3 Then after creating your `index.html`, `*.js`, `*.css`

`git add` all of the files from # 3
`git commit`
`git push origin gh-pages`

Also just remember you are always inside a tree, when working with git. Branches of the tree are either your branch names, or they can be hashes of the commits also.

**Comment by russelljjarvis** [comment]
If you also get a free account at travis, every time you push to git, it tries to build your code for free on the cloud, then the badge advertises to people looking at your git, that the code is build-able for everyone not just you.

Here is an example
https://travis-ci.org/russelljjarvis/neuronunit
neuronunit is my github code, it doesn't build because I am still developing it.

**Comment by TsaiHuiAnn** [comment]
Thank you Russell! I do try to "git branch" in the command but it didn't show up on the repository. Do you know why? I sometimes to uploaded or revise the code directly on the webpage, would that be the reason why it couldn't be synced?

**Comment by TsaiHuiAnn** [comment]
TsaiHuiAnn.github.io $ git branch
* homepage
  master

**Comment by TsaiHuiAnn** [comment]
Well..I finally push it...

**Comment by russelljjarvis** [comment]
Yes that type of error is very commonplace for standard git usage.

Its basically gits way of saying it what is online is ahead of whats on your local computer (you may think that its your local computer thats ahead, but usually they are both ahead, this is a complicated situation thats hard to explain). 

run: 
`
$git status
 ` 
to see what files you have changed.
`git add changed_files`
`git commit -m "boring message"`
`git pull origin master`
`git mergetool` (this may prompt you to select a merge tool, this is totally worthwhile, merging is necessary in complicated multi developer programming environments)
`git commit -m "completed the merge"`
Here is an important part:
`git push origin homepage`
You might accidentally be doing 
`git push origin master`,
but you need to replace master with your current branch name, master is just the default production branch name.

Also your branch name `homepage` absolutely must be `gh-pages` instead, the `gh-pages` is not an arbitrary name, its a way of telling GH to render index.html, as a real webpage, with your current choice of `homepage` the source code that constitutes `index.html` will only be displayed as code (it will look like text not like a webpage).

**Comment by russelljjarvis** [comment]
You can also just whipe your online work, and replace it with whats on your local machine with:
`git push -f origin homepage` (force, like forced replacement).
or you can do `git add files` 
`git commit -m "boring"`
`git stash`
`git pull origin master`
`git stash pop`
`git merge-tool`
`git commit -m 'boring 2'`
`git push origin homepage`
